src/spatialdata/_utils.py

```diff
@@ -311,3 +313,43 @@ def _error_message_add_element() -> None:
         "write_labels(), write_points(), write_shapes() and write_table(). We are going to make these calls more "
         "ergonomic in a follow up PR."
     )
+
+
+def _check_match_length_channels_c_dim(
+    data: DaskArray | DataArray | DataTree, c_coords: str | list[str], cls_dims: tuple[str]
+) -> list[str]:
+    """
+    Check whether channel names `c_coords` are of equal length to the `c` dimension of the data.
+
+    Parameters
+    ----------
+    data
+        The image array
+    c_coords
+        The channel names
+    cls_dims
+        The dimensions of the particular `ImageModel`
+
+    Returns
+    -------
+    c_coords
+        The channel names as list
+    """
+    c_index = cls_dims.index("c")
+    c_length = (
```

Comment:
I'd add a quick safety check that all the scales of the data-tree have the same channel dim.

Reply:
not relevant anymore as I had to switch to `get_axes_names`, which has the dispatch.